skiplang/prelude/src/stdlib/other/Base64.sk

```diff
@@ -64,103 +74,56 @@ fun decodeString(encoded: String): String {
 }
 
 private fun decode_(dbuf: mutable Array<UInt8>, chars: Vector<Char>): void {
-    dt = decodeTable;
-    i = 0;
-    j = 0;
-    while (i < chars.size() && j < dbuf.size()) {
-      (a, !i) = (dt[chars[i].code()], i + 1);
-      (b, !i) = (dt[chars[i].code()], i + 1);
-      (c, !i) = (dt[chars[i].code()], i + 1);
-      (d, !i) = (dt[chars[i].code()], i + 1);
-      invariant(a != -1 && b != -1, "Invalid Base64 data");
-      if (j < dbuf.size()) {
-        dbuf.set(j, UInt8::truncate(a.shl(2) + b.and(0x30).shr(4)));
-        !j = j + 1;
-      };
-      if (j < dbuf.size()) {
-        invariant(c != -1, "Invalid Base64 data");
-        dbuf.set(j, UInt8::truncate( b.and(0xf).shl(4) + c.and(0x3c).shr(2)));
-        !j = j + 1;
-      };
-      if (j < dbuf.size()) {
-        invariant(d != -1, "Invalid Base64 data");
-        dbuf.set(j, UInt8::truncate( c.and(0x3).shl(6) + d));
-        !j = j + 1;
-      };
+  dt = decodeTable;
+  i = 0;
+  j = 0;
+
+  while (i + 3 < chars.size() && j < dbuf.size()) {
+    aChar = chars[i];
+    bChar = chars[i+1];
+    cChar = chars[i+2];
+    dChar = chars[i+3];
+    if (aChar == '=' || bChar == '=') break void;
+
+    a = dt[aChar.code()];
+    b = dt[bChar.code()];
+    c = if (cChar == '=') -1 else dt[cChar.code()];
+    d = if (dChar == '=') -1 else dt[dChar.code()];
```

Comment:
Isn't `dt['='.code()]` equal to -1 already?

I mean: do we really need the special cases?